sos/plugins/mssql.py

```diff
@@ -0,0 +1,94 @@
+# Copyright (C) 2018 Red Hat, K.K., Takayoshi Tanaka <tatanaka@redhat.com>
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import os
+from sos.plugins import Plugin, RedHatPlugin
+
+
+class MsSQL(Plugin, RedHatPlugin):
+    """Microsoft SQL Server on Linux
+    """
+
+    plugin_name = "mssql"
+    profiles = ('services',)
+    packages = ('mssql-server',)
+
+    option_list = [
+        ('mssql_conf', 'SQL Server configuration file.', '',
+         '/var/opt/mssql/mssql.conf')
+    ]
+
+    def setup(self):
+        mssql_conf = self.get_option('mssql_conf')
+
+        # Pick error log file from mssql_conf.
+        # Expecting the following format
+        # ```
+        # [filelocation]
+        # errorlogfile = /var/opt/mssql/log
+        # [sqlagent]
+        # errorlogfile = /var/opt/mssql/log/sqlagentstartup.log
+        # [network]
+        # kerberoskeytabfile = /var/opt/mssql/secrets/mssql.keytab
+        # ```
+        # Default values are described in the document:
+        # https://docs.microsoft.com/en-us/sql/linux
+        # /sql-server-linux-configure-mssql-conf
+        # ?view=sql-server-linux-2017#mssqlconf-format
```

Comment:
Let's just drop this URL; interested parties can easily find it if they need it. It's just a plan INI file, so we can simply document the sections and keys that we care about.